### PR TITLE
Convert old-styled string formatting to f-string

### LINKED ISSUE #1484: f-string
Since it's Python 3.6 or go home now, shall we change all the `.format()` to f-string? For better readability and performance.

@pydanny @webyneter @jayfk what do you think?

### PR DESCRIPTION
**Pull request by sfdye**
Closes #1484 and #1494

This PR converts all the old-styled:
1. `%`
2. `str.format()` 

to the new f-string available from Python 3.6:
https://docs.python.org/3/whatsnew/3.6.html

### CODE REVIEW

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def ready(self):
 
 @app.task(bind=True)
 def debug_task(self):
-    print('Request: {0!r}'.format(self.request))  # pragma: no cover
+    print(f'Request: {self.request!r}')  # pragma: no cover
```
Comment (sfdye) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: `!r` still works in f-string
See: https://www.python.org/dev/peps/pep-3101/#explicit-conversion-flag

**Comment by sfdye** [comment]
Hmm, getting an interesting issue here. Travis is failing due to flake8 not passing the new f-string. 

Right now the error is:

> AttributeError: 'Checker' object has no attribute 'JOINEDSTR'

See https://github.com/PyCQA/pyflakes/pull/80

which can be fixed by update the following line to `flake8==3.5.0`
https://github.com/pydanny/cookiecutter-django/blob/a252830c3155eba27839a1b27bdd72f2ba02f3b3/tox.ini#L9

However, flake8 F405 will come back again as seen #631.

@pydanny @luzfcb @webyneter @browniebroke what do you guys think?

**Comment by browniebroke** [comment]
Looks like there is no way to avoid updating to a more recent flake8 here. 
We can disable F405 on a per line basis using `# noqa: F405`.

**Comment by sfdye** [comment]
@browniebroke This will result in a lot of ugly `# noqa: F405` added to each line that violates F405.

What I can think of to fix remaining issues after upgrading flake8 to 3.5.0:

1. Add `F405` to global
2. Add `local.py`, `test.py`, `production.py` to flake8 exclude file

What do you think?

@pydanny @browniebroke @luzfcb

**Comment by browniebroke** [comment]
I suggest to start by upgrading flake8 in this PR, so we can see how much how bad will be the ugliness?

There are tradeoffs in disabling per line, per file or globally, and right now it's hard (at least for me) to see where we stand. Seeing the flake8 failures will help make a decision.

**Comment by sfdye** [comment]
@browniebroke You may see the ugliness now :smile:

**Comment by browniebroke** [comment]
I was expecting a nicer output on the CI, but I was a bit too optimistic 😄 ... Nevermind. 

I've run flake8 manually on a project generated from your branch, and here it is:

```
./config/wsgi.py:29:1: E303 too many blank lines (3)
./config/settings/local.py:16:9: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/local.py:17:1: F405 'TEMPLATES' may be undefined, or defined from star imports: .base
./config/settings/local.py:23:14: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/local.py:31:17: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/local.py:46:1: F405 'MIDDLEWARE' may be undefined, or defined from star imports: .base
./config/settings/local.py:47:1: F405 'INSTALLED_APPS' may be undefined, or defined from star imports: .base
./config/settings/local.py:52:1: E402 module level import not at top of file
./config/settings/local.py:53:1: E402 module level import not at top of file
./config/settings/test.py:14:1: F405 'TEMPLATES' may be undefined, or defined from star imports: .base
./config/settings/test.py:20:14: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/test.py:56:1: F405 'TEMPLATES' may be undefined, or defined from star imports: .base
./config/settings/production.py:20:14: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:29:38: F405 'MIDDLEWARE' may be undefined, or defined from star imports: .base
./config/settings/production.py:39:34: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:41:31: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:46:23: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:55:17: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:58:1: F405 'INSTALLED_APPS' may be undefined, or defined from star imports: .base
./config/settings/production.py:68:21: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:69:25: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:70:27: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:94:22: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:96:24: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:97:16: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:102:24: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:103:30: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:111:1: F405 'TEMPLATES' may be undefined, or defined from star imports: .base
./config/settings/production.py:121:1: F405 'DATABASES' may be undefined, or defined from star imports: .base
./config/settings/production.py:121:24: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:122:1: F405 'DATABASES' may be undefined, or defined from star imports: .base
./config/settings/production.py:122:40: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:123:1: F405 'DATABASES' may be undefined, or defined from star imports: .base
./config/settings/production.py:127:21: F405 'env' may be undefined, or defined from star imports: .base
./config/settings/production.py:192:13: F405 'env' may be undefined, or defined from star imports: .base
```

**Comment by sfdye** [comment]
@browniebroke you got the gist 😂

**Comment by browniebroke** [comment]
If we re-import explicitly `env` from `base`, the number of errors left is manageable:
https://github.com/browniebroke/f-strings-cookiecutter/commit/36ac1310a95ae78c1daff6a3ae7fa6e09fabc9fa

**Comment by sfdye** [comment]
@browniebroke I will give that a shot.

File: tox.ini [hunk 2]
```diff
@@ -6,7 +6,7 @@ envlist = py36
 passenv = LC_ALL, LANG, HOME
 deps =
     binaryornot
-    flake8==2.5.5
+    flake8==3.5.0
```
Comment (webyneter) [inline tox.ini]: I suggest that we address that with a straightforward

```ini
deps = -rrequirements.txt
```

therefore closing #1494.

File: tox.ini [hunk 3]
```diff
@@ -5,7 +5,7 @@ envlist = py36
 [testenv]
 deps =
     binaryornot
-    flake8==2.5.5
+    flake8==3.5.0
```
Comment (webyneter) [inline tox.ini]: I suggest that we address that with a straightforward

```ini
deps = -rrequirements.txt
```

therefore closing #1494.

**Comment by webyneter** [comment]
:clap:

**Comment by webyneter** [comment]
@sfdye is this ready to be merged?

**Comment by sfdye** [comment]
@webyneter I think so, if you don't have other concerns.

File: {{cookiecutter.project_slug}}/config/settings/production.py [hunk 4]
```diff
@@ -11,16 +12,16 @@
 
 # DATABASES
 # ------------------------------------------------------------------------------
-DATABASES['default'] = env.db('DATABASE_URL')
-DATABASES['default']['ATOMIC_REQUESTS'] = True
-DATABASES['default']['CONN_MAX_AGE'] = env.int('CONN_MAX_AGE', default=60)
+DATABASES['default'] = env.db('DATABASE_URL')  # noqa F405
```
Comment (webyneter) [inline {{cookiecutter.project_slug}}/config/settings/production.py]: Didn't `from .base import env` solved the issue with '`env` might be undefined'?

Comment (sfdye) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: This is complaint for `DATABASES`

Comment (webyneter) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: Oh, OK.

**Comment by webyneter** [comment]
LGTM.

**Comment by luzfcb** [comment]
@webyneter 

```diff
-from .base import *  # noqa
+from .base import env
```

I think this is going to break the whole thing.

```python
from .base import *  # noqa
```
does django see only one file, and not many files, as expected by `DJANGO_SETTINGS_MODULE`

i'm +1 to revert to `from .base import *  # noqa` 

and see when this flake8 issue will be fixed https://gitlab.com/pycqa/flake8/issues/156

**Comment by webyneter** [comment]
@luzfcb what do you mean by 'break'?

**Comment by jayfk** [comment]
We are setting all the defaults in base. We need the star import, otherwise those settings won‘t be imported (or in this context - applied).

**Comment by webyneter** [comment]
But whait, the star import didn't go anywhere...

**Comment by webyneter** [comment]
At the moment that's what's in the `master`:

```python
from .base import *  # noqa
from .base import env
```

**Comment by luzfcb** [comment]
@webyneter @jayfk  wow... :flushed: is true, my mistake. Sorry for the noise. I made a little mess with github diff UI

**Comment by webyneter** [comment]
Fuf, I was getting a little worried :) No problem, @luzfcb.

**Comment by jayfk** [comment]
Ah, right. Haven‘t seen that myself. Maybe because we are on mobile?

**Comment by sfdye** [comment]
I was a bit worried too when I wake up with 10 emails on the same issue.

________________________________

**Pull request merged**